### DISCUSSION TITLE
Questions about Luajnatives

### DISCUSSION THREAD

**Discussion by ZYPyDoki** [gudzpoz/luajava#226] [Q&A]
Hello author

1.luajnatives Is there any tutorial on how to use it? The text tutorial is indeed too complicated. Is there any video demonstration on how to use such content?

2.luajnatives Is it possible to load and use C modules in Luaj environment? Similar to Androula's require import using package.loadlib to load C modules, how can I use it? For example, the Luaj official website states that when loading a Lua file in the loadfile, if 'require' is used in this Lua file, cpath will not be loaded by default, only Lua or Java will be loaded. In addition, the loadlib of the package in the Luaj official version forces a nil empty string prompt, which is what I am puzzled about. How can I implement loading C modules in Luaj?

3.Perhaps my idea is quite imaginative. Luaj is a Lua environment implemented in Java, and it may be unrealistic to use Clua modules. However, I still want to ask if it is possible to implement it?

**Comment by gudzpoz** [comment]
1. If you mean [`LuaJNatives`](https://gudzpoz.github.io/luajava/javadoc/party/iroiro/luajava/luaj/LuaJNatives.html), the API follows the [Lua C API](https://www.lua.org/manual/5.1/manual.html#3) and I would recommend reading the manual (or find any suitable Lua C API tutorials, video or textual).
2. (& 3.) No. Lua C modules require real Lua C API, which only those non-Java implementation provides.
